torch/ao/quantization/utils.py

```diff
@@ -370,3 +373,105 @@ def has_no_children_ignoring_parametrizations(module):
         return len(module._modules) == 1 and 'parametrizations' in module._modules
     else:
         return False
+
+def _get_path_of_module(root, submodule):
+    """ Get the path (fully qualified name) of a submodule
+    """
+    for n, p in root.named_modules():
+        if submodule is p:
+            return n
+
+def _get_signature_locals(f, loc):
+    """ Get local keyword arguments
+    """
+    return {k: v for k, v in loc.items() if k in signature(f).parameters}
+
+def _get_default_kwargs(f):
+    """ Get all default keyword arguments from function signature
+    """
+    kwargs = {}
+    for name, param in signature(f).parameters.items():
+        if param.default is not param.empty:
+            kwargs[name] = param.default
+        elif param.kind is param.VAR_POSITIONAL:
+            kwargs[name] = ()
+        elif param.kind is param.VAR_KEYWORD:
+            kwargs[name] = {}
+    return OrderedDict(kwargs)
+
+def _normalize_kwargs(func, loc):
+    """ Given a function and local function arguments, normalize the keyword
+    arguments by filling in default arguments from function signature
+    """
+    default_kwargs = _get_default_kwargs(func)
+    local_kwargs = _get_signature_locals(func, loc)
+    normalized_kwargs = default_kwargs.copy()
+    for attr, val in local_kwargs.items():
+        if attr in normalized_kwargs:
+            # override the default keyword arguments
+            normalized_kwargs[attr] = val
+    return normalized_kwargs
+
+def _get_num_pos_args(f):
+    """ Get number of positional args for a function
+    """
+    return len(getfullargspec(f).args)
+
+def get_fqn_to_example_inputs(model, example_inputs):
+    """ Given a model and its example inputs, return a dictionary from
+    fully qualified name of submodules to example_inputs for that submodule,
+    e.g. {"linear1": (tensor1,), "linear2": (tensor2,), "sub": (tensor3,),
+          "sub.linear1": (tensor4,), ...}
+
+    Used to make quantizing submodules easier now that FX Graph Mode Quantization requries
+    example inputs.
+
+    Also works for keyword arguments with default values, we would flatten keyword
+    arguments as positional arguments and fill in the missing keyword args with default
+    values, e.g. if we have a forward function:
+    def forward(self, x, key1=3, key2=3):
+        ...
+
+    and we call it with self.submodule(x, key=6)
```

Comment:
should this say `key2` instead of `key`?

Reply:
oh yes, this should be `key2`